core/runtime/src/main/java/io/quarkus/runtime/annotations/RegisterForReflection.java

```diff
@@ -29,6 +29,13 @@
      */
     boolean ignoreNested() default true;
 
+    /**
+     * Whether the full class hierarchy should be processed or not
+     *
+     * This is intended to be used to register types that are going to be serialized
+     */
+    boolean ignoreHierarchy() default true;
```

Comment:
Wondering if this is useful? I can't remember a use case where you want to ignore the hierarchy, because you need the full hierarchy of the class to use it. If you ignore it then you need to register the missing pieces manually.

Reply:
I totally agree. I've put that in order to keep backward compatibility regarding current behavior. It's may be unlikely, but this change can cause cascading software behavior changes, for those who have coded expecting the current behavior.

Reply:
You need the inherited fields otherwise you will lose information within serialization.

Reply:
Ok but in most cases when we register for reflection we don't do it for Serialization purposes?

Reply:
Let's remember that each reflective registration can potentially trigger a significant overhead, as the compiler can't fold type cardinality individually in each callsite: a reflective type registration is global and can potentially appear anywhere.

So let's try to nudge towards being more selective rather than more inclusive - when it makes sense.

Reply:
Are you saying that you prefer to add a way to control if registration should happen for parent classes?

Reply:
yes I think it's useful and that it should default to `true` for the general case.

Perhaps if the intended use is serialization we should have a different build item to make the intent clearer? Then for the serialization case you could remove this flag and always include parents.

Reply:
> Perhaps if the intended use is serialization we should have a different build item to make the intent clearer? 


@Sanne, do you mean the implementation could be better by creating a BuildStep and a BuildItem specific for that use case and also a annotation?

Something like:
RegisterForSerializationBuildStep
RegisterForSerializationBuildItem
RegisterForSerialization (annotation)

For this case the default would be full hierarchy processing.

Or the RegisterForReflectionBuildStep could be reused for this new BuildItem? If that makes any sense.

Reply:
yes right I'm wondering if that makes sense. However while I'm familiar with reflection, I don't have experience with Serialization in native-image so don't take my word for it: I'm just wondering if that makes sense, based on your comments as it seems one would want slightly different processing & defaults.

This way you could avoid having an attribute for `ignoreHierarchy`: it would always be off for reflection, and always on for serialization to provide better APIs.

Reply:
I can implement that alternative and submit another pr, referencing this one, so we can compare both approaches. 

Is that ok?